blockmanager.go

```diff
@@ -3044,3 +3091,167 @@ func (l *lightHeaderCtx) RelativeAncestorCtx(
 		ancestorHeight, ancestor, l.store, l.headerList,
 	)
 }
+
+func (b *blockManager) sideLoadHeaders() {
+	// If sideloading is not enabled on this chain
+	// return quickly.
+	if !b.cfg.sideLoad.Enabled {
```

Comment:
No need to even call at all if this isn't enabled. All this logic can be skipped.

Reply:
If it is no enabled, it returns quickly no logic here.